packages/react-codemods/transforms/pf3-pf4.button.test.js

```diff
@@ -1,9 +1,29 @@
+import { EOL as SYSTEM_EOL } from 'os';
 import prettier from 'prettier';
 import { defineInlineTest } from 'jscodeshift/dist/testUtils';
 import transform from './pf3-pf4';
 
-const prettierConfig = prettier.resolveConfig.sync(process.cwd());
-const pretty = src => prettier.format(src, { parser: 'babylon', ...prettierConfig });
+/**
+ * Codemod outputs should follow the EOL pattern of the target codebase, not
+ * Patternfly's. Patternfly currently enforces LF as line ending, independent of
+ * the OS building the codebase.
+ *
+ * JSCodeShift produces OS-dependent line endings however, LF for Unix based
+ * systems, and CRLF for Windows based ones. This is also likely what most
+ * projects would like to see the codemod prduce.
+ *
+ * To make sure we both adhere to Patternfly's conventions, and compare the
+ * correct OS-specific line endings during the test cases' assertions, we store
+ * expected values with LF line endings, and convert them into the OS-specific
+ * ones at runtime using prettier.
+ */
+const PRETTIER_EOL = SYSTEM_EOL === '\r\n' ? 'crlf' : 'cr';
```

Comment:
With all the extra effort it takes to work around line ending efforts, maybe it would be a better approach to disable the line ending lint rule, and let Git ensure that they are correctly set?

If Git's `core.autocrlf` option is set correctly, it should make sure that all files have LF line endings after check-in, but devs can work with the OS-specific line endings locally.

There are likely ways to force non-LF endings into the code base with this setup, but that risk may be a worthy tradeoff for the added flexibility?

Reply:
I agree.  @tlabaj @jschuler Thoughts?

Reply:
@dana Thoughts?